src/memory/server/MemoryServer.ts

```diff
@@ -40,13 +37,14 @@ export class MemoryServer {
                 retainContextWhenHidden: true,
             }
         );
-        context.subscriptions.push(this.panel);
+        context.subscriptions.push(newPanel);
 
-        this.panel.webview.onDidReceiveMessage((message) =>
-            this.onDidReceiveMessage(message)
-        );
+        newPanel.webview.onDidReceiveMessage((message) => {
+            this.panel = newPanel;
```

Comment:
This looks like you are using this global to pass the `newPanel` to where it needs to go. Instead of having a global (field) simply pass `newPanel` to `onDidReceiveMessage` and onwards to everywhere that needs this extra context.